GUI/ppm-gui

```diff
@@ -823,13 +641,293 @@ class ApplicationWindow(QtWidgets.QMainWindow):
+        vb = QtWidgets.QHBoxLayout(); bxr.addLayout(vb)
+        self.bxSSx = _spin_box((1, 100), 16, 0.1, self.updateScanWindow, TTips['ScanSize'], vb)
+        self.bxSSy = _spin_box((1, 100), 16, 0.1, self.updateScanWindow, TTips['ScanSize'], vb)
+
+        # Scan start
+        lb = QtWidgets.QLabel("Scan start (x,y)[Å]"); lb.setToolTip(TTips['ScanStart']); bxl.addWidget(lb)
+        vb = QtWidgets.QHBoxLayout(); bxr.addLayout(vb)
+        self.bxSCx = _spin_box((-100, 100), 0, 0.1, self.updateScanWindow, TTips['ScanStart'], vb)
+        self.bxSCy = _spin_box((-100, 100), 0, 0.1, self.updateScanWindow, TTips['ScanStart'], vb)
+
+        # Distance, amplitude, rotation
+        vb = QtWidgets.QHBoxLayout(); self.l0.addLayout(vb)
+        lb = QtWidgets.QLabel("Distance [Å]"); lb.setToolTip(TTips['Distance']); vb.addWidget(lb)
+        self.bxD = _spin_box((-1000, 1000), 6.5, 0.1, self.updateScanWindow, TTips['Distance'], vb)
+        lb = QtWidgets.QLabel("Amplitude [Å]"); lb.setToolTip(TTips['Amplitude']); vb.addWidget(lb)
+        self.bxA = _spin_box((0.0, 100), 1.0, 0.1, self.updateScanWindow, TTips['Amplitude'], vb)
+        lb = QtWidgets.QLabel("Rotation"); lb.setToolTip(TTips['Rotation']); vb.addWidget(lb)
+        self.bxRot = _spin_box((-360.0, 360.0), 0.0, 5.0, self.updateRotation, TTips['Rotation'], vb)
+
+    def _create_pbc_settings_ui(self):
+
+        # Title
+        vb = QtWidgets.QHBoxLayout(); self.l0.addLayout(vb)
+        lb = QtWidgets.QLabel("Periodic Boundaries"); lb.setAlignment(QtCore.Qt.AlignCenter)
+        font = lb.font(); font.setPointSize(12); lb.setFont(font); lb.setMaximumHeight(50)
+        vb.addWidget(lb)
+
+        # Toggle PBC
+        vb = QtWidgets.QHBoxLayout(); self.l0.addLayout(vb)
+        lb = QtWidgets.QLabel("Use periodic boundary conditions"); lb.setToolTip(TTips['PBC']); vb.addWidget(lb)
+        self.bxPBC = QtWidgets.QCheckBox()
+        self.bxPBC.setChecked(True)
+        self.bxPBC.toggled.connect(self.updatePBC)
+        vb.addWidget(self.bxPBC)
+
+        # Toggle z PBC
+        lb = QtWidgets.QLabel("z periodicity"); lb.setToolTip(TTips['PBCz']); vb.addWidget(lb)
+        self.bxPBCz = QtWidgets.QCheckBox()
+        self.bxPBCz.setChecked(False)
+        self.bxPBCz.toggled.connect(self.updatePBC)
+        self.bxPBCz.setToolTip(TTips['PBCz'])
+        vb.addWidget(self.bxPBCz)
+
+        # Left-right divide for labels and input boxes
+        vb = QtWidgets.QHBoxLayout(); self.l0.addLayout(vb)
+        bxl = QtWidgets.QVBoxLayout(); vb.addLayout(bxl, 1)
+        bxr = QtWidgets.QVBoxLayout(); vb.addLayout(bxr, 3)
+
+        # Lattice vector A
+        lb = QtWidgets.QLabel("A (x, y, z) [Å]"); lb.setToolTip(TTips['PBC']); bxl.addWidget(lb)
+        vb = QtWidgets.QHBoxLayout(); bxr.addLayout(vb)
+        self.bxPBCAx = _spin_box((-100, 100), 50, 0.1, self.updatePBC, TTips['PBC'], vb)
```

Comment:
Let's hope that no one will have bigger systems than 100angstrom in near future!

Reply:
Well, it's probably pretty rare to go above 100 Å, but I could just increase it up to 1000 Å.

Reply:
I do not think so for now, on most of nowadays machines, such a big FF would lead to _memory overflow_ ...

Reply:
There could be cases where the periodic cell is big in one direction, but narrow in another, like some ribbons of 2D materials. Also, the lattice vector size here does not affect the memory usage. I think pyqt just requires the maximum values to determine the minimum size of the spin box. So there should be no harm in increasing the maximum value.